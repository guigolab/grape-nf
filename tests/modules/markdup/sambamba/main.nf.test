nextflow_process {

    name "Test Process markdup"
    script "modules/markdup/sambamba/main.nf"
    process "markdup"

    tag "module"

    test("Should mark duplicates using sambamba") {

        when {
            process {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
