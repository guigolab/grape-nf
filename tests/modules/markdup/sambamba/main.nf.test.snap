{
    "Should mark duplicates using sambamba": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.markdup.bam:md5,c720dedc6b6d94a7a507543fd44b1e3a",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.markdup.bam.bai:md5,9696c48348b5e0d26d4c45a211fc9bdd",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ],
                "dedupedAlignments": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.markdup.bam:md5,c720dedc6b6d94a7a507543fd44b1e3a",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "dedupedAlignmentsIndices": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.markdup.bam.bai:md5,9696c48348b5e0d26d4c45a211fc9bdd",
                        "bai",
                        "GenomeAlignmentsIndex",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T19:38:12.840625"
    }
}